truffle/src/main/java/org/jruby/truffle/runtime/rope/LeafRope.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2015, 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.runtime.rope;
+
+import org.jcodings.Encoding;
+
+import java.util.Arrays;
+
+public abstract class LeafRope extends Rope {
+
+    private final byte[] bytes;
```

Comment:
Should be @CompilationFinal and not just final (so it indicates very byte is final as well).